### PR TITLE
DOC: fixed Ex03 errors in docstrings: 

### PR DESCRIPTION
**Pull request by tiffanyxiao**
**PR SUMMARY**

Checked if validation docstrings passes for:

- [x] pandas.errors.DatabaseError
- [x] pandas.errors.IndexingError 
- [x] pandas.errors.InvalidColumnName

OUTPUT:
1. python scripts/validate_docstrings.py --format=actions --errors=EX03 pandas.errors.DatabaseError
```
################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.errors.DatabaseError" correct. :)
```

2. python scripts/validate_docstrings.py --format=actions --errors=EX03 pandas.errors.IndexingError
```
################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.errors.IndexingError" correct. :)
```

3. python scripts/validate_docstrings.py --format=actions --errors=EX03 pandas.errors.InvalidColumnName
```
################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.errors.InvalidColumnName" correct. :)
```

**PR CHECKLIST**

- [ ] xref #56804 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/errors/__init__.py [hunk 1]
```diff
@@ -599,19 +599,30 @@ class IndexingError(Exception):
     """
     Exception is raised when trying to index and there is a mismatch in dimensions.
 
+    Raised by properties like :attr:`.pandas.DataFrame.iloc` when
+    an indexer is out of bounds or :attr:`.pandas.DataFrame.loc` when its index is
+    unalignable to the frame index.
+
+    See Also
+    --------
+    :attr:`.pandas.DataFrame.iloc` : Purely integer-location based indexing for \
```
Comment (datapythonista) [inline pandas/errors/__init__.py:608]: I don't think in the `See Also` section we need those `:attr:` markers, I think just `DataFrame.iloc` should render correctly. Same for the other cases in this PR.

**Review by datapythonista — COMMENTED** [review]
Thanks @tiffanyxiao, great job. Just one comment, as I think you can simplify the formatting in the see also section.

Also, there seems to be a conflict after merging another PR. Can you merge main into your PR and resolve the conflict please?

File: pandas/errors/__init__.py (hunk 1 above)
Comment (tiffanyxiao) [reply pandas/errors/__init__.py]: Hi Marc, 

Thanks! I pushed these changes up.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @tiffanyxiao

**Pull request merged**